control-plane/src/main.rs

```diff
@@ -160,36 +164,62 @@ async fn tcp_server() -> Result<()> {
         .ok()
         .and_then(|raw_data_plane_version| semver::Version::parse(&raw_data_plane_version).ok());
 
+    let sem = Arc::new(Semaphore::new(MAX_REQS_MINUTE));
+    let permit_creator = sem.clone();
+
+    tokio::spawn(async move {
+        loop {
+            let to_add = MAX_REQS_MINUTE.checked_sub(permit_creator.available_permits());
+
+            StatsClient::record_requests_minute(to_add.unwrap_or(0) as f64);
+
+            if let Some(to_add) = to_add {
+                permit_creator.add_permits(to_add);
+            }
+
+            sleep(Duration::from_secs(60)).await;
+        }
+    });
```

Comment:
This loop shouldn't be needed - we can handle the aggregation in our metrics by publishing a counter for requests received as they come in

Reply:
Although it may be good to publish a metric on total + available file descriptors from the control plane

Reply:
I agree this isn't be the way I think we should implement this if it was just for logging but the idea was to be able to just be able to change the number of max permits when we want to actually start rate limiting?

So this logic is really intended to be used for rate limiting as opposed to logging. Understand if you think it's better to remove this until a future date though